src/main/java/bio/terra/service/resourcemanagement/ResourceService.java

```diff
@@ -48,21 +48,39 @@ public ResourceService(
         GoogleProjectService projectService,
         GoogleBucketService bucketService,
         SamConfiguration samConfiguration,
-        DatasetBucketDao datasetBucketDao,
-        GoogleResourceDao resourceDao) {
+        DatasetBucketDao datasetBucketDao) {
         this.dataLocationSelector = dataLocationSelector;
         this.projectService = projectService;
         this.bucketService = bucketService;
         this.samConfiguration = samConfiguration;
         this.datasetBucketDao = datasetBucketDao;
-        this.resourceDao = resourceDao;
+    }
+
+    /**
+     * Fetch/create a project
+     *
+     * @param dataset
+     * @param billingProfile authorized profile for billing account information case we need to create a project
+     * @return a reference to the project as a POJO GoogleProjectResource
+     */
+    public GoogleProjectResource getOrCreateProjectForBucket(Dataset dataset,
+                                                             BillingProfileModel billingProfile)
+        throws GoogleResourceException, GoogleResourceNamingException, InterruptedException {
+
+        final GoogleProjectResource datasetProject = getProjectResource(dataset.getProjectResourceId());
+        String sourceDatasetGoogleProjectId = datasetProject.getGoogleProjectId();
+        final GoogleRegion region = dataset.getDatasetSummary().getStorageResourceRegion(GoogleCloudResource.FIRESTORE);
+        // Every bucket needs to live in a project, so we get or create a project first
+        return projectService.getOrCreateProject(
+            dataLocationSelector.projectIdForFile(dataset, sourceDatasetGoogleProjectId, billingProfile),
+            billingProfile,
+            null,
```

Comment:
is this supposed to be null?

Reply:
Thanks for looking so closely at this!
After looking into the code a bit, I think this was copied from the original "getOrCreateBucketForFile" method. This variable is the roleIdentityMapping argument, which I think is only used for projects created for datasets.